### PR TITLE
Add gen-docs scripts

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This pr import gen-docs script to automatically generate documentation for karmadactl and karmada components based on cobra refer to https://github.com/kubernetes/kubernetes/tree/master/cmd/gendocs.

Our current cli tools are still in a rapid development stage, and these generated docs are not intended to be our user-facing documentation directly, but only for reference.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```
Users can go build the gen-docs files and run the output binary to generate the markdown files.
```

### CODE REVIEW

**Comment by Poor12** [comment]
/cc @RainbowMango @lonelyCZ 
You can see the effect by https://deploy-preview-206--karmada.netlify.app/docs/next/reference/karmadactl/karmadactl-commands/karmadactl_index.
BTW, index page is generated by the long description of the command. Content of some commands still need to improve.

File: cmd/gendocs/gen_karmadactl_docs.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+		lines[len(lines)-1] = "#### Go Back to [Karmadactl Commands](karmadactl_index) Homepage.\n\n\n###### Auto generated by [spf13/cobra script in Karmada](https://github.com/karmada-io/karmada/tree/master/cmd/gendocs)."
+
+		title := strings.TrimPrefix(firstL, "## ")
+		lines[0] = "---"
+		newlines := []string{"---", "title: " + title}
+
+		newlines = append(newlines, lines...)
+		newcontent := strings.Join(newlines, "\n")
+		return ioutil.WriteFile(path, []byte(newcontent), info.Mode())
```
Comment (RainbowMango) [inline cmd/gendocs/gen_karmadactl_docs.go:141]: The `WriteFile` has been deprecated, please use os.WriteFile instead.

File: cmd/gencomponentdocs/postprocessing.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// MarkdownPostProcessing goes though the generated files
```
Comment (RainbowMango) [inline cmd/gencomponentdocs/postprocessing.go:11]:
```suggestion
// MarkdownPostProcessing goes through the generated files
```
typo

File: pkg/karmadactl/util/types.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package util
```
Comment (RainbowMango) [inline pkg/karmadactl/util/types.go:1]: I think we can rename the file to `commandgroup.go` or `command_group.go`.
`types.go` is usually used to define API types.

**Review by RainbowMango — COMMENTED** [review]
Generally looks good, just a small suggestion.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**